app/views/downtimes/_form.html.erb

```diff
@@ -7,26 +7,67 @@
 <% end %>
 
 <div class="form-group form-inline js-start-time">
-  <%= f.label :start_time, 'This service will be unavailable from' %><br />
-  <span>
-  <%= f.time_select :start_time, { default: 1.hour.from_now.beginning_of_hour, ignore_date: true }, { class: 'date form-control' } %>
-</span>
+  <p>This service will be unavailable from</p>
+  <div class="date-label">
+    <%= f.label :start_time, "Hour", for: "date_start_time_4i" %>
+    <%= select_hour @downtime.start_time&.hour || 1.hour.from_now.beginning_of_hour, { field_name: "start_time(4i)" }, { class: "form-control date" } %>
+  </div>
+
+  <div class="date-label">
+    <%= f.label :start_time, "Minute", for: "date_start_time_5i" %>
+    <%= select_minute @downtime.start_time&.minute, { field_name: "start_time(5i)" }, { class: "form-control date" } %>
+  </div>
+
   <b class="add-left-margin add-right-margin">on</b>
-  <%= f.date_select :start_time, { order: [:day, :month, :year], default: Date.tomorrow,
-    start_year: Date.today.year, end_year: Date.today.year.next }, { class: 'date form-control' } %>
+
+  <div class="date-label">
+    <%= f.label :start_time, "Day", for: "date_start_time_3i" %>
+    <%= select_day @downtime.start_time&.day || Date.tomorrow.day, { field_name: "start_time(3i)" }, { class: "form-control date" } %>
+  </div>
+
+  <div class="date-label">
+    <%= f.label :start_time, "Month", for: "date_start_time_2i" %>
+    <%= select_month@downtime.start_time&.month || Date.tomorrow.month, { field_name: "start_time(2i)" }, { class: "form-control date" } %>
+  </div>
+
+  <div class="date-label">
+    <%= f.label :start_time, "Year", for: "date_start_time_1i" %>
+    <%= select_year @downtime.start_time&.year || Date.tomorrow.year, { field_name: "start_time(1i)" }, { class: "form-control date" } %>
+  </div>
 </div>
 
 <div class="form-group form-inline js-stop-time">
-  <%= f.label :end_time, 'to' %><br />
-  <span>
-  <%= f.time_select :end_time, { default: 1.hour.from_now.beginning_of_hour, ignore_date: true }, { class: 'date form-control' } %>
-  </span>
+  <p>to</p>
+
+  <div class="date-label">
+    <%= f.label :end_time, "Hour", for: "date_end_time_4i" %>
+    <%= select_hour @downtime.end_time&.hour || 1.hour.from_now.beginning_of_hour, { field_name: "end_time(4i)" }, { class: "form-control date" } %>
```

Comment:
Having a form object would also help with setting these values as the form could have a default value on initialisation which would be called in the new controller action 
```
def new 
  @downtime = DowntimeForm.new 
end
```

and then a build_from_downtime method for edit 

```
def edit
  @downtime = DowntimeForm.build_from_downtime(Downtime.find(params[:id]
end 

class DowntimeForm

attr_accessor :all_the_accessors

def build_from_downtime(downtime)
  new(
    message: downtime.message
    etc.
  )
end
```